docker/archive/multi-writer.go

```diff
@@ -0,0 +1,51 @@
+package archive
+
+import (
+	"context"
+
+	"github.com/containers/image/v5/types"
+	"github.com/pkg/errors"
+)
+
+// MultiImageArchiveReader allows for creating and writing to a docker-archive
+// with multiple images.  Once created, it can be used in subsequent copy.Image
+// operations when set as the copy.Options.ImageDestination.
+//
+// For using a MultiImageArchiveWriter in copy.Image, set it as the
+// copy.Option.ImageDestination.
+type MultiImageArchiveWriter struct {
+	*archiveImageDestination
+}
+
+func NewMultiImageArchiveWriter(ctx context.Context, sys *types.SystemContext, path string) (*MultiImageArchiveWriter, error) {
+	ref := archiveReference{path: path}
+	dst, err := newArchiveImageDestination(sys, ref)
```

Comment:
I don’t think this works at all; how does the user specify tags for the individual images?

Again, let this object primarily track the temporary file (and more state about layers, in that case), and give out `ImageReference` objects that can be used to write a single image.

Reply:
Tags can be specified via `AddRepoTags()`. I just finished rewriting the source parts and will now look into the destination.